--- ../Src/code/ColladaHelper.h	Fri Feb 03 05:38:32 2012
+++ ../Src/code/ColladaHelper.h	Sun Jan 20 11:35:08 2013
@@ -228,6 +228,8 @@
   std::string mSID;
 	Node* mParent;
 	std::vector<Node*> mChildren;
+	
+	aiNodeType mNodeType;
 
 	/** Operations in order to calculate the resulting transformation to parent. */
 	std::vector<Transform> mTransforms;
@@ -250,6 +252,7 @@
 	//! Constructor. Begin with a zero parent
 	Node() { 
 		mParent = NULL;
+		mNodeType = aiNodeType_Unknown;
 	}
 
 	//! Destructor: delete all children subsequently
@@ -265,6 +268,13 @@
 	bool mIsStringArray;
 	std::vector<float> mValues;
 	std::vector<std::string> mStrings;
+	std::string mFullString;
+
+	Data()
+	{
+		mIsStringArray = false;
+		mFullString = "";
+	}
 };
 
 /** Accessor to a data array */
