--- ColladaHelper.h.orig	2017-12-11 14:22:45.000000000 +0100
+++ ColladaHelper.h	2018-01-09 03:11:29.166871800 +0100
@@ -256,6 +256,8 @@
     Node* mParent;
     std::vector<Node*> mChildren;
 
+    aiNodeType mNodeType;
+
     /** Operations in order to calculate the resulting transformation to parent. */
     std::vector<Transform> mTransforms;
 
@@ -277,6 +279,7 @@
     //! Constructor. Begin with a zero parent
     Node() {
         mParent = NULL;
+		mNodeType = aiNodeType_Unknown;
     }
 
     //! Destructor: delete all children subsequently
@@ -292,6 +295,13 @@
     bool mIsStringArray;
     std::vector<ai_real> mValues;
     std::vector<std::string> mStrings;
+    std::string mFullString;
+
+    Data()
+    {
+        mIsStringArray = false;
+        mFullString = "";
+    }
 };
 
 /** Accessor to a data array */
