--- ./scene.h.orig	2017-12-11 14:22:45.000000000 +0100
+++ ./scene.h	2018-01-09 03:55:57.912166700 +0100
@@ -121,6 +121,8 @@
       */
     unsigned int* mMeshes;
 
+    aiNodeType mNodeType;
+
     /** Metadata associated with this node or NULL if there is no metadata.
       *  Whether any metadata is generated depends on the source file format. See the
       * @link importer_notes @endlink page for more information on every source file
