--- types.h.orig	2017-12-11 14:22:45.000000000 +0100
+++ types.h	2018-01-09 03:56:49.263259300 +0100
@@ -449,6 +449,23 @@
     /// @endcond
 }; // !enum aiDefaultLogStream
 
+typedef enum aiNodeType
+{
+	aiNodeType_Unknown = 0x0,
+	aiNodeType_MeshController = 0x1,
+	aiNodeType_Mesh = 0x2,
+	aiNodeType_Light = 0x3,
+	aiNodeType_Camera = 0x4,
+	aiNodeType_Bone = 0x5,	
+
+	//Like a Biped Object
+	aiNodeType_BoneHelper = 0x6,
+
+
+	
+	_AI_NT_ENFORCE_ENUM_SIZE = 0x7fffffff 
+} aiNodeType;
+
 // just for backwards compatibility, don't use these constants anymore
 #define DLS_FILE     aiDefaultLogStream_FILE
 #define DLS_STDOUT   aiDefaultLogStream_STDOUT
